androidApp/src/main/java/com/mbta/tid/mbta_app/android/stopDetails/DepartureTile.kt

```diff
@@ -0,0 +1,202 @@
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.layout.onSizeChanged
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.semantics.heading
+import androidx.compose.ui.semantics.selected
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import com.mbta.tid.mbta_app.android.MyApplicationTheme
+import com.mbta.tid.mbta_app.android.R
+import com.mbta.tid.mbta_app.android.component.RoutePill
+import com.mbta.tid.mbta_app.android.component.RoutePillType
+import com.mbta.tid.mbta_app.android.util.fromHex
+import com.mbta.tid.mbta_app.android.util.modifiers.haloContainer
+import com.mbta.tid.mbta_app.model.ObjectCollectionBuilder
+import com.mbta.tid.mbta_app.model.RealtimePatterns
+import com.mbta.tid.mbta_app.model.RouteType
+import com.mbta.tid.mbta_app.model.TripInstantDisplay
+
+@Composable
+fun DepartureTile(
+    data: TileData,
+    onTap: () -> Unit,
+    modifier: Modifier = Modifier,
+    showRoutePill: Boolean = false,
+    showHeadsign: Boolean = true,
+    isSelected: Boolean = false
+) {
+    Column(
+        modifier
+            .clickable(
+                onClickLabel =
+                    if (isSelected) null
+                    else stringResource(R.string.departure_tile_on_click_label),
+                onClick = onTap
+            )
+            .heightIn(min = 56.dp)
+            .haloContainer(
+                borderWidth = 2.dp,
+                outlineColor = if (isSelected) colorResource(R.color.halo) else Color.Transparent,
+                backgroundColor =
+                    if (isSelected) colorResource(R.color.fill3)
+                    else colorResource(R.color.deselected_toggle_2).copy(alpha = 0.6f),
+                borderRadius = 8.dp
+            )
+            .padding(10.dp)
+            .semantics {
+                if (isSelected) {
+                    heading()
+                    selected = true
+                }
+            },
+        verticalArrangement = Arrangement.spacedBy(4.dp, Alignment.CenterVertically)
+    ) {
+        CompositionLocalProvider(
+            LocalContentColor provides
+                if (isSelected) colorResource(R.color.text)
+                else colorResource(R.color.deselected_toggle_text)
+        ) {
+            // If you can figure out a better way to make the pill left aligned to the text and the
+            // status right aligned to the text whether the text is wider or narrower than the pill
+            // + status, please do.
```

Comment:
Messing around with it a bit locally, I think `width(IntrinsicSize.Max)` on the Column and `fillMaxWidth()` on the headsign text will have the effect you're looking for

Reply:
That did it, thanks! I don't think I'd used `IntrinsicSize` before; I'll definitely have to remember that.